dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/scheduling/DefaultSchedulingManager.java

```diff
@@ -101,21 +103,28 @@ public DefaultSchedulingManager( JobService jobService, JobConfigurationService
     @Override
     public void schedule( JobConfiguration configuration )
     {
-        log.info( String.format( "Scheduling job: %s", configuration ) );
+        boolean isCronTriggered = configuration.getJobType().getSchedulingType() == SchedulingType.CRON;
+        if ( isCronTriggered && isNullOrEmpty( configuration.getCronExpression() ) )
+        {
+            log.info(
+                format( "Ignoring CRON schedule for %s as it is not ready to run.", configuration.getJobType() ) );
+            return;
+        }
+        log.info( format( "Scheduling job: %s", configuration ) );
```

Comment:
With `@Slf4j` we can use `{}` to indicate variables and skip `String.format` entirely.